api/handlers_e2e_test.go

```diff
@@ -300,6 +302,98 @@ func TestPagingRootTxnDeduplication(t *testing.T) {
 	}
 }
 
+func TestTransactionHandler(t *testing.T) {
+	db, shutdownFunc := setupIdb(t, test.MakeGenesis(), test.MakeGenesisBlock())
+	defer shutdownFunc()
+
+	///////////
+	// Given // A block containing a key reg txn with state proof key
+	///////////
+	var votePK crypto.OneTimeSignatureVerifier
+	votePK[0] = 1
+
+	var selectionPK crypto.VRFVerifier
+	selectionPK[0] = 1
+
+	var sprfkey merklekeystore.Verifier
+	sprfkey[0] = 1
+
+	txn := transactions.SignedTxnWithAD{
+		SignedTxn: transactions.SignedTxn{
+			Txn: transactions.Transaction{
+				Type: "keyreg",
+				Header: transactions.Header{
+					Sender:      test.AccountA,
+					GenesisHash: test.GenesisHash,
+				},
+				KeyregTxnFields: transactions.KeyregTxnFields{
+					VotePK: votePK,
+					SelectionPK:selectionPK,
+					StateProofPK: sprfkey,
+					VoteFirst: basics.Round(0),
+					VoteLast: basics.Round(100),
+					VoteKeyDilution: 1000,
+					Nonparticipation: false,
+				},
+			},
+			Sig: test.Signature,
+		},
+	}
+
+	block, err := test.MakeBlockForTxns(test.MakeGenesisBlock().BlockHeader, &txn)
+	require.NoError(t, err)
+
+	err = db.AddBlock(&block)
+	require.NoError(t, err, "failed to commit")
+
+	//////////
+	// When // We query the txn
+	//////////
+
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.SetPath("/v2/transactions/:txid")
+	api := &ServerImplementation{db: db}
+	//params := generated.LookupAccountByIDParams{}
+	err = api.LookupTransaction(c, txn.Txn.ID().String())
+	require.NoError(t, err)
+	require.Equal(t, http.StatusOK, rec.Code)
+
+	//////////
+	// Then // The key reg txn response has state proof key
+	//////////
+
+	var response generated.TransactionResponse
+	data := rec.Body.Bytes()
+	err = json.Decode(data, &response)
+	require.NoError(t, err)
+	require.NotNil(t, response.Transaction.KeyregTransaction.StateProofKey)
+	require.Equal(t, json.Encode(sprfkey), json.Encode(*response.Transaction.KeyregTransaction.StateProofKey))
```

Comment:
This is interesting. Is there a reason to compare json encodings instead of byte arrays?

Reply:
sprfKey type merklekeystore.Verifier and response.Transaction.KeyregTransaction.StateProofKey is deserialized to *byte[].  This check fails if I don't convert them to the same type.

Reply:
Casting both to `[]byte` would be a simpler option

Reply:
it seems that I'm only allowed to cast sprfKey to [64]byte because merklekeystore.Verifier is a type alias for [64]byte.

Reply:
Oh good point. Will `sprfkey[:]` to convert to a slice?

Reply:
updated.